File: software/examples/csharp/ExampleThreshold.cs
using System;
using Tinkerforge;

class Example
{
	private static string HOST = "localhost";
	private static int PORT = 4223;
	private static string UID = "XYZ"; // Change XYZ to the UID of your Ozone Bricklet

	// Callback function for ozone concentration reached callback (parameter has unit ppb)
	static void OzoneConcentrationReachedCB(BrickletOzone sender, int ozoneConcentration)
	{
		Console.WriteLine("Ozone Concentration: " + ozoneConcentration + " ppb");
	}

	static void Main()
	{
		IPConnection ipcon = new IPConnection(); // Create IP connection
		BrickletOzone o = new BrickletOzone(UID, ipcon); // Create device object

		ipcon.Connect(HOST, PORT); // Connect to brickd
		// Don't use device before ipcon is connected

		// Get threshold callbacks with a debounce time of 10 seconds (10000ms)
		o.SetDebouncePeriod(10000);

		// Register ozone concentration reached callback to function OzoneConcentrationReachedCB
		o.OzoneConcentrationReachedCallback += OzoneConcentrationReachedCB;

		// Configure threshold for ozone concentration "greater than 20 ppb" (unit is ppb)
		o.SetOzoneConcentrationCallbackThreshold('>', 20, 0);

		Console.WriteLine("Press enter to exit");
		Console.ReadLine();
		ipcon.Disconnect();
	}
}

File: software/examples/javascript/ExampleThreshold.js
var Tinkerforge = require('tinkerforge');

var HOST = 'localhost';
var PORT = 4223;
var UID = 'XYZ'; // Change to your UID

var ipcon = new Tinkerforge.IPConnection(); // Create IP connection
var o = new Tinkerforge.BrickletOzone(UID, ipcon); // Create device object

ipcon.connect(HOST, PORT,
    function (error) {
        console.log('Error: ' + error);
    }
); // Connect to brickd
// Don't use device before ipcon is connected

ipcon.on(Tinkerforge.IPConnection.CALLBACK_CONNECTED,
    function (connectReason) {
        // Get threshold callbacks with a debounce time of 10 seconds (10000ms)
        o.setDebouncePeriod(10000);

        // Configure threshold for ozone concentration "greater than 20 ppb" (unit is ppb)
        o.setOzoneConcentrationCallbackThreshold('>', 20, 0);
    }
);

// Register ozone concentration reached callback
o.on(Tinkerforge.BrickletOzone.CALLBACK_OZONE_CONCENTRATION_REACHED,
    // Callback function for ozone concentration reached callback (parameter has unit ppb)
    function (ozoneConcentration) {
        console.log('Ozone Concentration: ' + ozoneConcentration + ' ppb');
    }
);

console.log('Press key to exit');
process.stdin.on('data',
    function (data) {
        ipcon.disconnect();
        process.exit(0);
    }
);
